src/main/java/org/primeframework/mvc/action/config/DefaultActionConfigurationBuilder.java

```diff
@@ -136,7 +137,13 @@ public ActionConfiguration build(Class<?> actionClass) {
     Field unknownParametersField = findUnknownParametersField(actionClass);
     Set<String> validContentTypes = findAllowedContentTypes(actionClass);
 
-    return new ActionConfiguration(actionClass, allowKnownParameters, constraintValidationMethods, executeMethods, validationMethods, formPrepareMethods, authorizationMethods, jwtAuthorizationMethods, postValidationMethods, preParameterMethods, postParameterMethods, resultAnnotations, preParameterMembers, preRenderMethodsMap, fileUploadMembers, memberNames, securitySchemes, scopeFields, additionalConfiguration, uri, preValidationMethods, unknownParametersField, validContentTypes);
+    MessageResources messageResources = actionClass.getAnnotation(MessageResources.class);
+    String fallbackMessageResourcesPath = null;
+    if (messageResources != null) {
+      fallbackMessageResourcesPath = uriBuilder.build(messageResources.fallback());
```

Comment:
The only potential issue with looking this up by action at start is that there are some complex lookup options. 

For example: `/api/{id}/foo/{id}` - but maybe this is ok?  since we'll use the action to resolve the resource, and this is not the same as resolving the action from the URI which is what I am describing. And if it doesn't work for 100% of the use cases, we'll find it during dev time I suppose. 

But the alternative would be to have the annotation take in strings such as `/api/foo`, and this then doesn't get caught by refactors.

Reply:
I think it's an asset of PMVC to rely on the typing, so I prefer that over the string. It could get a little complex, yes. But one should kind of expect that when they use this annotation, no?